### PR TITLE
fix(appointments): create and manage recurring bookings atomically

### PR DESCRIPTION
**Pull request by yingbull**
The R dialog on a saved appointment offers only Recurring Update/Cancel/Delete. On A12, Recurring Update reports a successful addition but creates no repeats. The old JSP also updates the first date repeatedly, archives the wrong row when cancelling, accepts invalid/unbounded ranges and can leave a partially saved series.

Share one validated form and transactional service between the new-booking and saved-appointment R routes. Add an explicit Create repeats action for saved appointments, with guidance to save edits first, a future end-date default, visible validation errors and accurate affected-row counts. Update/Cancel/Delete operate on existing members through the selected end date. Move mutations into a transactional service; validate the entire range before writes, lock the existing series, preserve its legacy identity and creation metadata, archive each changed/deleted occurrence, and avoid duplicate inserts when the same saved appointment is submitted again. Month/year generation retains the original day; management also includes old month-end dates that drifted under the former calendar logic. Operations are bounded to 366 appointments and ten years. No schema or Flyway migration is introduced.

Add a registered Playwright workflow using the existing UI booking fixture (importing that fixture no longer executes the standalone lifecycle test): book from a schedule slot → R → invalid-date refusal/correction → Create repeats → repeat submission → Recurring Update/Cancel/Delete. Verify exact weekly dates, patient/provider/times/notes/reason, no duplicate rows, and an archive for every affected occurrence. Cleanup removes only this run's fixtures.

Validation:
- Reproduced the false-success/no-repeats behavior on checksum-verified published A12 DEBs on Ubuntu 26.04, through actual UI clicks and database checks.
- 27 unit/real-DAO/routing/security-gate tests pass: inclusive ranges, invalid intervals, range bounds, month/leap-year behavior, existing-appointment expansion/idempotence, all-row update/cancel/delete/archive, unsaved edits, permissions, legacy shifted dates, non-patient bookings, text provider IDs, write-only new-booking permissions, and a simulated second-insert failure that proves transaction rollback.
- All 664 Node regressions pass, including all 18 manifest checks. JavaScript syntax and diff checks pass. The local sandbox blocked child-process tests with EPERM; the full suite passed sequentially outside it.
- Packaged latest release/2026.08 plus this change; the final VM UI workflow passes for saved-patient recurrences and new blocked-time bookings. The new-booking route also creates three blocked-time appointments without a patient. Builds ran only with the VM stopped, a 2 GiB Java heap and a 10 GiB host-memory reserve. Browser runs use one browser at a time on the 6 GiB VM.

Based on and targets release/2026.08, including concurrent merged fixes through 928102dacb. DCO signed.

Fixes #3705

Review also reconciled concurrently added suite entries. The documentation uses the live suite listing instead of fragile hand-maintained totals; the smoke timeout ceiling and manifest/script/alias completeness are tested directly.

## Summary by Sourcery

Make recurring appointment creation and management atomic, validated, idempotent, and consistent across new and saved bookings.

New Features:
- Add a shared recurring-booking workflow that creates repeats for both new and saved appointments, including explicit saved-appointment creation and management actions.

Bug Fixes:
- Prevent false-success recurring operations, duplicate bookings, incorrect series targeting, partial writes, and missed legacy month-end occurrences.

Enhancements:
- Centralize recurrence validation and operations in a transactional service with locking, bounded ranges, metadata preservation, per-occurrence archiving, and accurate result counts.
- Improve the recurring-booking UI with validation feedback, guidance, retained form values, future end-date defaults, and localized labels.

CI:
- Register an end-to-end Playwright recurrence workflow and strengthen suite checks for smoke timeout budgets and manifest coverage.

Documentation:
- Document the recurring appointment Playwright workflow and replace fragile hand-maintained suite totals with live listings.

Tests:
- Add unit, integration, routing, and Playwright coverage for recurrence ranges, calendar edge cases, idempotence, permissions, metadata, archiving, and transaction rollback.

Chores:
- Route the legacy recurring-booking page through the shared recurring form without introducing a schema migration.

---
## Summary by cubic
Builds recurring appointment creation and management on a single validated form and transactional service, so saved bookings no longer report success without creating repeats and no recurring operation leaves a partially saved series. Fixes #3705.

**Recurring bookings**
- Saved appointments get an explicit Create repeats action; Update/Cancel/Delete operate on existing series members through the selected end date.
- The whole range validates and locks before writes, repeated submission creates no duplicates, and every changed or deleted occurrence is archived.
- Ranges are bounded to 366 appointments and ten years, and month/year generation preserves the original day, including legacy-drifted month-end dates.
- `style`/`billing` values absent from the edit form are kept per occurrence unless explicitly cleared.
- No schema or Flyway migration is introduced.

**Testing**
- Adds a Playwright workflow covering the full recurrence lifecycle, plus unit/integration tests for bounds, leap/month-end behavior, idempotence, permissions, metadata preservation, and transaction rollback.

<sup>Written for commit 1914fcd5be2c56b956fdf0657056d5c813d4cd2e. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/carlos-emr/carlos/pull/3742?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

**Comment by yingbull** [comment]
@coderabbitai full review

File: src/main/java/io/github/carlos_emr/carlos/appointment/service/RecurringAppointmentService.java [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+        a.setDemographicNo(demographic.isBlank() ? 0 : number(demographic, "patient number"));
+        a.setProgramId(programId);
+        a.setName(v.getOrDefault("keyword", ""));
+        a.setNotes(v.getOrDefault("notes", ""));
+        a.setReason(v.getOrDefault("reason", ""));
+        a.setLocation(v.getOrDefault("location", ""));
+        a.setResources(v.getOrDefault("resources", ""));
+        a.setType(v.getOrDefault("type", ""));
+        a.setStyle(v.getOrDefault("style", ""));
+        a.setBilling(v.getOrDefault("billing", ""));
```
[the remark this answers is not in the document]
Comment (yingbull) [reply src/main/java/io/github/carlos_emr/carlos/appointment/service/RecurringAppointmentService.java:153]: Addressed in c5e955483c, with schema-sized fixtures in 81e85a195e. Saved-repeat creation inherits omitted style/billing from the anchor; group updates preserve each occurrence's own omitted values. Explicitly submitted empty strings can still clear them. Added real-DAO tests for both paths and explicit clearing (29 focused tests pass). The new VM Playwright regression reproduced the old rejection, then passed creation/update/cancel/delete and blocked-time creation with the fixed runtime; it asserts style/billing survive the actual edit form.

File: src/main/webapp/WEB-INF/jsp/appointment/appointmenteditrepeatbooking.jsp [hunk 2]
```diff
@@ -417,40 +236,40 @@
...
                         <input type="hidden" name="everyUnit" id="everyUnit"
                                class="form-control form-control-sm" style="width: 8rem;"
-                               value="day" readonly>
+                               value="<%= SafeEncode.forHtmlAttribute(request.getParameter("everyUnit") == null ? "day" : request.getParameter("everyUnit")) %>" readonly>
 
                         </div>
                         <div class="form-check form-check-inline">
                             <input class="form-check-input" type="radio" name="dateUnit" id="dateUnitDay"
-                                   value="day" checked onclick='onCheck(this, "day")'>
+                                   value="day" <%= request.getParameter("everyUnit") == null || "day".equals(request.getParameter("everyUnit")) ? "checked" : "" %> onclick='onCheck(this, "day")'>
```
[the remark this answers is not in the document]
Comment (yingbull) [reply src/main/webapp/WEB-INF/jsp/appointment/appointmenteditrepeatbooking.jsp]: The flagged expression could emit only the fixed token checked or an empty string; no request value was emitted there. In c5e955483c the selection inputs are read once before rendering, making the fixed-token comparisons explicit. The actual everyUnit hidden value remains HTML-attribute encoded, as do forwarded parameters. Semgrep PRO passed on c5e955483c; the full VM recurrence workflow also passed.

**Comment by yingbull** [comment]
@coderabbitai full review
